_resources/port1.0/group/gpg_verify-1.0.tcl

```diff
@@ -61,6 +61,6 @@ proc gpg_verify.verify_gpg_signature {pubkey_file signature_file test_file} {
             then echo 'VERIFIED'; else echo 'UNVERIFIED'; fi"]
     if {[string trim ${gpg_verification}] != "VERIFIED"} {
         ui_error "GPG signature verification failed on ${test_file} with pubkey file ${pubkey_file}."
-        exit 1
+        error "GPG signature verification failed on ${test_file} with pubkey file ${pubkey_file}."
```

Comment:
No need to repeat the same message with ui_error and error.

Reply:
> No need to repeat the same message with ui_error and error.

But I’ve observed that there is a good reason: without an explicit `exit`, port commands continue after a failed verification, undermining the very purpose of this port group.

I’m not convinced that removing the explicit `exit` Is a good idea, but don’t know enough about MacPorts error handling to suggest a workable alternate.

I just know that when I tested this with a known bad signature, a failed verification don’t exit as expected.

Reply:
I just read https://github.com/macports/macports-ports/pull/5124#discussion_r361985931:

To be clear, `ui_error` doesn’t throw an error? That would explain the behavior I describe immediately above.

All that’s necessary is `error “Message”`, correct?

Reply:
Changed to just `error`.

BTW, what’s the purpose of the `ui_error` command?

Reply:
Yes, that looks fine. `ui_error` is for displaying an error message to the user in whatever way the client deems appropriate. With the command-line client, that's printing it to stderr with an "Error: " prefix. A GUI client might pop up a dialog instead. You don't always want to break control flow and send an error up the call stack whenever you do that.